### DISCUSSION TITLE
Custom Curved Text

### DISCUSSION THREAD

**Discussion by Udhayan26** [fabricjs/fabric.js#7873] [Q&A]
Hello everybody,

I really need some help on my custom curved text.

I like this type of code [https://codepen.io/mrIee/pen/gEKaxp](https://codepen.io/mrIee/pen/gEKaxp)

But this code working in fabric version: '5.2.0' 

Any ideas how I can fix that? 

Thank you so much

Udhayan

**Comment by melchiar** [comment]
Seems to work fine with v5.2.1 (version 5.2.0 didn't publish correctly, hence the quick release of 5.2.1)

[https://codepen.io/melchiar/pen/MWrqpyX](https://codepen.io/melchiar/pen/MWrqpyX)

**Reply by DineshCodeFlow** [reply]
Try scaling any object. it stuttering or hangs the browser!

**Comment by timoostrich** [comment]
The performance of that example is very bad, especially when it comes to bigger canvas sizes.
It sets the dimensions of the object by searching for colored pixels. You should not do that and better calculate the needed width/height for the object. That's way faster!

**Comment by maxwellfrolov** [comment]
Hello community. 
Is there a way to create CurvedText in several lines (up to 5)?

**Comment by vavilon224** [comment]
@melchiar  @covationlabs @Udhayan26  Hello everyone, how can I integrate it into my project ?? React.js + Fabric 5.2.1

It doesn't work consistently error :(
